info-compiler/src/test/java/com/google/sps/NewsContentExtractorTest.java

```diff
@@ -77,33 +87,51 @@ public void extractContentFromHtml_realValidWebpageWithWrongUrlParam() {
+    Assert.assertTrue(potentialNewsArticle.isPresent());
+    NewsArticle newsArticle = potentialNewsArticle.get();
+    Assert.assertEquals(newsArticle, NEWS_ARTICLE_WITH_WRONG_URL);
+  }
+
+  @Test
+  public void extractContentFromHtml_emptyDocParser() throws IOException, SAXException,
+      TikaException {
+    // Extract content and meta data using a parser that modifies the content handler to
+    // receive an end-of-document notification via {@code endDocument()}. Therefore, an empty
+    // article will be extracted, while the URL is set correctly to the passed in parameter.
+    // Content processing hasn't occurred so the abbreivated content is null.
+    HtmlParser parser = mock(HtmlParser.class);
+    doAnswer(parseInvocation -> {
+               BoilerpipeContentHandler boilerpipeHandler =
+                   (BoilerpipeContentHandler) parseInvocation.getArguments()[1];
+               boilerpipeHandler.endDocument();
+               return null;
+            }).when(parser).parse(anyObject(), anyObject(), anyObject());
+    NewsContentExtractor newsContentExtractor = new NewsContentExtractor(parser);
+    Optional<NewsArticle> potentialNewsArticle =
+        newsContentExtractor.extractContentFromHtml(realWebpageStream,
+                                                    NEWS_ARTICLE_WITH_EMPTY_CONTENT.getUrl());
     Assert.assertTrue(potentialNewsArticle.isPresent());
     NewsArticle newsArticle = potentialNewsArticle.get();
-    Assert.assertEquals(newsArticle.getTitle(), NEWS_ARTICLE.getTitle());
-    Assert.assertNotEquals(newsArticle.getUrl(), NEWS_ARTICLE.getUrl());
-    Assert.assertEquals(newsArticle.getUrl(), WRONG_URL);
-    Assert.assertTrue(newsArticle.getContent().contains(NEWS_ARTICLE.getContent()));
-    Assert.assertNull(newsArticle.getAbbreviatedContent());
+    Assert.assertEquals(newsArticle, NEWS_ARTICLE_WITH_EMPTY_CONTENT);
   }
 
   @Test
-  public void extractContentFromHtml_alwaysValidWebpage() throws IOException {
-    // Extract content and meta data from an always valid webpage that contains no content. The
-    // extraction process is correct, and extracted title and content are {@code EMPTY}. URL is
-    // set correctly to {@code URL}. Content processing hasn't occurred so the abbreivated content
-    // is null.
-    InputStream alwaysValidStream = mock(InputStream.class);
-    when(alwaysValidStream.read(anyObject(), anyInt(), anyInt()))
+  public void extractContentFromHtml_alwaysValidAndEmptyWebpageStream() throws IOException {
+    // Extract content and meta data from an always valid and empty webpage that contains no
+    // content. The extraction process is correct, and extracted title and content are {@code
+    // EMPTY}. URL is set correctly to {@code URL}. Content processing hasn't occurred so the
+    // abbreivated content is null.
+    InputStream alwaysValidAndEmptyStream = mock(InputStream.class);
```

Comment:
No need to mock the InputStream anymore if you're mocking the parser anyways (and can return the textdocument you want)